File: 백준/Silver/15652. N과 M （4）/N과 M （4）.py
import sys
input = sys.stdin.readline

def back(x):
    if len(answer) == m:
        print(*answer)
        return

    for i in range(x, n+1):
        answer.append(i)
        back(i)
        answer.pop()

n, m = map(int, input().split())
answer = []
back(1)

File: 백준/Silver/15652. N과 M （4）/N과 M （4）.swift
import Foundation

let input = readLine()!.split(separator: " ").map { Int($0)! }
let n = input[0], m = input[1]
var answer = Array(repeating: -1, count: m)

combi(0, 0)

func combi(_ idx: Int, _ before: Int) {
    if idx == m {
        for i in 0..<m { print(answer[i], terminator: " ") }
        print()
        return
    }

    for i in before..<n {
        answer[idx] = i + 1
        combi(idx + 1, i)
    }
}